src/utils/fhirFilter.ts

```diff
@@ -92,3 +92,58 @@ export const addCancerHistologyMorphology = (
   }
   return condition;
 };
+
+export function convertStringtoResource({
+  bundle,
+  valueString,
+  id,
+  profile_value,
+  codingSystem,
+  codingSystemCode,
+}: {
+  bundle: Bundle;
+  valueString: string;
+  id: string;
+  profile_value: string;
+  codingSystem: string;
+  codingSystemCode: string;
+}): void {
+  // Create the Condition - done separate from the function call to ensure proper TypeScript checking
+  let code: CodeableConcept = null;
+  let resource: Observation;
+  if (codingSystemCode) {
+    code = {
+      coding: [
+        {
+          system: codingSystem,
+          code: codingSystemCode,
+        },
+      ],
+    };
+  }
+  if (code) {
+    const resource: Observation = {
```

Comment:
`code` can only be `null` if `codingSystemCode` is falsey. Is there any particular reason to have a separate conditional?